tests/ui/excessive_precision.rs

```diff
@@ -66,4 +72,10 @@ fn main() {
 
     // issue #7745
     let _ = 1.000_000_000_000_001e-324_f64;
+
+    // issue #9910
+    const INF1: f32 = 1.0e+33f32;
+    const INF2: f64 = 1.0e+3300f64;
+    const NEG_INF1: f32 = -1.0e+33f32;
+    const NEG_INF2: f64 = -1.0e+3300f64;
```

Comment:
It seems like these literals does not cover the overflowing issue.

I tried using
```rust
const X: f32 = -3.40282357e+38_f32;
```

and it still fired and fixed it to:
```rust
const X: f32 = -inf_f32;
```